mass_mailing_custom_unsubscribe/templates/mass_mailing_contact_reason.xml

```diff
@@ -0,0 +1,46 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<!-- Copyright 2016 Jairo Llopis <jairo.llopis@tecnativa.com>
+     License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl). -->
+
+<odoo>
+
+    <template id="unsubscribe" inherit_id="mass_mailing.unsubscribe" name="Add Reasons to Mailing List Management Form">
+        <!-- Disable core AJAX submission of form, because it is impossible to
+             extend it as it is designed right now. -->
+        <xpath expr="//div[hasclass('container', 'o_unsubscribe_form')]" position="attributes">
+            <attribute name="class" value="container o_unsubscribe_form_custom"/>
```

Comment:
IMHO it makes more sense to just disable upstream JS file in the assets definition. Remove this part and see below.